site/git-gpg.md

```diff
@@ -0,0 +1,247 @@
+
+1. Retrieve the key ID (second row of the `sec` column):
+
+   ```sh
+   $  gpg --list-secret-keys --keyid-format long <email address>
+   sec   rsa2048/6A7436E8790F8689 2019-03-28 [SC]
+         700D397C988079BFF0DDAFED6A7436E8790F8689
+   uid                 [ultimate] Weaveworks Flux <support@weave.works>
+   ssb   rsa2048/ECA4FF5BD988B8E9 2019-03-28 [E]
+   ```
+
+2. Export the public and private keypair from your local GPG keyring
+   to a Kubernetes secret with `--export-secret-keys <key id>`:
+
+   ```sh
+   $ gpg --export-secret-keys 700D397C988079BFF0DDAFED6A7436E8790F8689 |
+     kubectl create secret generic flux-gpg-keys --from-file=flux.asc=/dev/stdin --dry-run -o yaml
+   apiVersion: v1
+   data:
+     flux.asc: <base64 string>
+   kind: Secret
+   metadata:
+     creationTimestamp: null
+     name: flux-gpg-keys
+
+3. Adapt your Flux deployment to mount the secret and enable the
+   signing of commits:
+
+   ```yaml
+   spec:
+     template:
+        spec:
+          volumes:
+          - name: gpg-keys
+              secret:
+                secretName: flux-gpg-keys
+                defaultMode: 0400
+          containers:
+          - name: flux
+            volumeMounts:
+            - name: gpg-keys
+              mountPath: /root/gpg-import
+              readOnly: true
+            args:
+            - --git-gpg-key-import=/root/gpg-import
+            - --git-signing-key=700D397C988079BFF0DDAFED6A7436E8790F8689 # key id
+   ```
+
+   or set the `gpgKeys.secretName` in your Helm `values.yaml` to
+   `gpg-keys`, and `signingKey` to your `<key id>`.
+
+4. To validate your setup is working, run `git log --show-signature` or
+   `git verify-tag <configured label>` to assure Flux signs its git
+   actions.
+
+   ```sh
+   $ git verify-tag <configured label>
+   gpg: Signature made vr 29 mrt 2019 15:28:34 CET
+   gpg:                using RSA key 700D397C988079BFF0DDAFED6A7436E8790F8689
+   gpg: Good signature from "Weaveworks Flux <support@weave.works>" [ultimate]
+   ```
+
+> **Note:** Flux *does not* recursively scan a given directory but does
+understand symbolic links to files.
+
+> **Note:** Flux will automatically add any imported key to the GnuPG
+  trustdb. This is required as git will otherwise not trust signatures
+  made with the imported keys.
+
+# Signature verification
+
+The verification of commit signatures is enabled by importing all
+trusted public keys (`--git-gpg-key-import=<path>`), and by setting the
+`--gpg-verify-signatures` flag. Once enabled Flux will verify all
+commit signatures, and the signature from the sync tag it is comparing
+revisions with.
+
+In case a signature can not be verified, Flux will sync state up to the
+last valid revision it can find _before_ the unverified commit was
+made, and lock on this revision.
+
+## Importing trusted GPG keys and enabling verification
+
+1. Collect the public keys from all trusted git authors.
+
+2. Add the collected keys to your secret with GPG keys.
```

Comment:
It is not super obvious how to do this step. Would it make it easier if `--gpg-import-keys` was slice-valued, then one could create a new secret from a directory of public keys, and mount it in its own directory. This would be easier to explain how to do, and easier to do, I reckon.

Reply:
And the public keys could live in a `ConfigMap` instead of a `Secret` which also makes more sense. Will make adjustments.

Reply:
I tried this after your latest commit, and it works very nicely